misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -526,7 +539,36 @@ private StackFrame toDapStackFrame(StackFrameProxyImpl stackFrame) {
             dapStackFrame.setSource(source);
             dapStackFrame.setLine((long) stackFrame.location().lineNumber());
             dapStackFrame.setColumn(0L);
-            dapStackFrame.setName(stackFrame.location().method().name());
+
+            if (stackFrame.location().method().name().matches(WORKER_LAMBDA_REGEX)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) == null) {
+                Value strand = ((ArrayReference) stackFrame.getStackFrame().getArgumentValues().get(0))
+                        .getValue(0);
+                String stackFrameName = String.valueOf(((ObjectReference) strand)
+                        .getValue(((ObjectReference) strand).referenceType().fieldByName(NAME)));
+                stackFrameName = removeRedundantQuotes(stackFrameName);
+                dapStackFrame.setName(WORKER + FRAME_SEPARATOR + stackFrameName);
+            } else if (stackFrame.location().method().name().contains(LAMBDA)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) == null) {
+                Value strand = ((ArrayReference) stackFrame.getStackFrame().getArgumentValues().get(0))
+                        .getValue(0);
+                String stackFrameName = String.valueOf(((ObjectReference) strand)
+                        .getValue(((ObjectReference) strand).referenceType().fieldByName(NAME)));
+                stackFrameName = removeRedundantQuotes(stackFrameName);
+                if (stackFrameName.equals(NULL)) {
+                    stackFrameName = ANONYMOUS;
+                }
+                dapStackFrame.setName(START + FRAME_SEPARATOR + stackFrameName);
+            } else if (stackFrame.location().method().name().contains(LAMBDA)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) != null) {
+                Value strand = stackFrame.getValue(stackFrame.visibleVariableByName(STRAND_VAR_NAME));
+                String stackFrameName = String.valueOf(((ObjectReference) strand)
+                        .getValue(((ObjectReference) strand).referenceType().fieldByName(NAME)));
+                stackFrameName = removeRedundantQuotes(stackFrameName);
+                dapStackFrame.setName(stackFrameName);
+            } else {
+                dapStackFrame.setName(stackFrame.location().method().name());
+            }
```

Comment:
IMO better to move this logic into a separate method `getStackFrameName()`. WDYT?

Reply:
Also lets add error recovery for the above util method so that the method will return a default name(`anonymous` maybe?) , if any exception is occurred while computing the stack frame name.

Reply:
Added to a separate method `getStackFrameName()` and added error recovery to this util.